pkg/operator/apiservercontrollerset/apiservercontrollerset.go

```
@@ -0,0 +1,179 @@
+) *APIServerControllerSet {
+	nodeProvider := DaemonSetNodeProvider{
+		TargetNamespace:                  cs.targetNamespace,
+		TargetNamespaceDaemonSetInformer: kubeInformersForNamespaces.InformersFor(cs.targetNamespace).Apps().V1().DaemonSets(),
+		NodeInformer:                     kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes(),
+	}
+
+	deployer, err := encryptiondeployer.NewRevisionLabelPodDeployer(
+		"revision",
+		cs.targetNamespace,
+		kubeInformersForNamespaces,
+		resourceSyncer,
+		cs.kubeClient.CoreV1(),
+		cs.kubeClient.CoreV1(),
+		nodeProvider,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	migrator := migrators.NewInProcessMigrator(dynamicClientForMigration, cs.kubeClient.Discovery())
+
+	cs.encryptionControllers, err = encryption.NewControllers(
+		cs.targetNamespace,
+		deployer,
+		migrator,
+		cs.operatorClient, // or should it be apiserverOperatorClient
+		configClient.ConfigV1().APIServers(),
+		configInformers.Config().V1().APIServers(),
+		kubeInformersForNamespaces,
+		cs.kubeClient.CoreV1(),
+		cs.eventRecorder,
+		encryptionResources...,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	cs.revisionController = revisioncontroller.NewRevisionController(
+		cs.targetNamespace,
+		nil,
+		[]revision.RevisionResource{{
+			Name:     "encryption-config",
+			Optional: true,
+		}},
+		kubeInformersForNamespaces.InformersFor(cs.targetNamespace),
+		OpenshiftDeploymentLatestRevisionClient{OperatorClient: cs.operatorClient, TypedClient: operatorConfigClient.OperatorV1()},
+		v1helpers.CachedConfigMapGetter(cs.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		v1helpers.CachedSecretGetter(cs.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		cs.eventRecorder,
+	)
+
+	cs.pruneController = prune.NewPruneController(
+		cs.targetNamespace,
+		[]string{"encryption-config-"},
+		cs.kubeClient.CoreV1(),
+		cs.kubeClient.CoreV1(),
+		kubeInformersForNamespaces,
+		cs.eventRecorder,
+	)
+
+	return cs
+}
+
```

add a check method where you can return an error for an unhandled (no with or without) controller.

---

> add a check method where you can return an error for an unhandled (no with or without) controller.

Still needed.  Use the trick we used in k8s.io/apiserver for .PrepareRun().Run() with types to ensure someone has to call and see if there are new controllers.

---

done